test/functional/JIT_Test/src/jit/test/recognizedMethod/TestMathUtils.java

```diff
@@ -0,0 +1,301 @@
+            double b = r.nextDouble() * 1000000000.0d;
+            int sign1 = r.nextInt(2) == 0 ? -1 : 1;
+            int sign2 = r.nextInt(2) == 0 ? -1 : 1;
+            numList[i][0] = a * sign1;
+            numList[i][1] = b * sign2;
+        }
+        return numList;
+    }
+
+    private static NaNTestPair<Double, Long> getDoubleNaNPair(){
+        NaNTestPair<Double, Long> p = new NaNTestPair<Double, Long>();
+        double farg1, farg2;
+        farg1 = farg2 = 0.0f;
+
+        long iarg1;
+        long min = 1L << 13;
+        do {
+            iarg1 = r.nextLong();
+        } while (iarg1 < min);
+        // shift right so that mantisa is non-zero
+        // shift of 13 instead of 12 ensures that the quiet bit is not set
+        iarg1 = iarg1 >> 13;
+
+        int arg1Type = r.nextInt(3); // 0 -> quiet, 1 -> signalling
+        if (arg1Type == 0) {
+            iarg1 = iarg1 | dNaNExpStart | dquietBit;
+            farg1 = Double.longBitsToDouble(iarg1);
+        } else if (arg1Type == 1) {
+            iarg1 = iarg1 | dNaNExpStart;
+            farg1 = Double.longBitsToDouble(iarg1);
+        } else {
+            // Normal number (non NaN)
+            farg1 = r.nextDouble() * 1000000.0 + 1.0;
+        }
+
+        // arg 2
+        long iarg2;
+        do {
+            iarg2 = r.nextLong();
+            iarg2 = iarg2 >> 13;
+        } while (iarg2 < min);
+
+        int arg2Type = r.nextInt(3);
+        if (arg2Type == 0) {
+            iarg2 = iarg2 | dNaNExpStart | dquietBit;
+            farg2 = Double.longBitsToDouble(iarg2);
+        }
+        else if (arg2Type == 1) {
+            iarg2 = iarg2 | dNaNExpStart;
+            farg2 = Double.longBitsToDouble(iarg2);
+        } else {
+            if (arg1Type != 2) {
+                farg2 = r.nextDouble() * 1000000.0 + 1;
+                iarg2 = Double.doubleToRawLongBits(farg2);
+            } else {
+                return null;
+            }
+        }
+
+        p.first = farg1;
+        p.bFirst = iarg1;
+        p.second = farg2;
+        p.bSecond = iarg2;
+        p.expected = (arg1Type != 2 ? iarg1 : iarg2);
+        return p;
+    }
+
+    private static NaNTestPair<Float, Integer> getFloatNaNPair(){ // 0: max, 1: min
+        NaNTestPair<Float, Integer> p = new NaNTestPair<Float, Integer>();
+        float farg1, farg2;
+        farg1 = farg2 = 0.0f;
+
+        int iarg1 = r.nextInt(fNaNMantisaMax - 1) + 1;
+        int arg1Type = r.nextInt(3); // 0 -> quiet, 1 -> signalling
+        if (arg1Type == 0) {
+            iarg1 = iarg1 | fNaNExpStart | fquietBit;
+            farg1 = Float.intBitsToFloat(iarg1);
+        } else if (arg1Type == 1) {
+            iarg1 = iarg1 | fNaNExpStart;
+            farg1 = Float.intBitsToFloat(iarg1);
+        } else {
+            // Normal number (non NaN)
+            farg1 = r.nextFloat() * 1000000.0f + 1.0f;
+            message = message + "(" + String.valueOf(farg1) + "f, ";
```

Comment:
You forgot to clean this one - this has caused the functional tests I launched fail. Can you fix this and launch the sanitytests on internal jenkins and share link on slack with me? Once they finish, I will launch test again and then merge.